grpc/server/server.go

```diff
@@ -0,0 +1,135 @@
+package server
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/credentials"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	log "github.com/sirupsen/logrus"
+	"github.com/soheilhy/cmux"
+
+	"net"
+
+	"github.com/gorilla/mux"
+	"github.com/keratin/authn-server/app"
+	"github.com/keratin/authn-server/grpc/internal/errors"
+	"github.com/keratin/authn-server/grpc/private"
+	"github.com/keratin/authn-server/grpc/public"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+)
+
+func init() {
+	runtime.HTTPError = errors.CustomHTTPError
+}
+
+// RunPrivateService starts a gRPC server for the private API and accompanying gRPC-Gateway server
+func RunPrivateService(ctx context.Context, app *app.App, grpcListener net.Listener, httpListener net.Listener) error {
```

Comment:
this func does not appear to be used outside the package. minimizing exported functions helps with code comprehension. would you review other packages as well?

```suggestion
func runPrivateService(ctx context.Context, app *app.App, grpcListener net.Listener, httpListener net.Listener) error {
```